docs/source/guides/tools_menu.rst

```diff
@@ -0,0 +1,201 @@
+
+For more details or to add a new board, see the `boards.txt`_ file.
+
+Generic Options
+---------------
+
+Most of the options are available for every ESP32 families. Some options will be available only for specific targets, like the USB configuration.
+
+Board
+*****
+
+This option is the target board and must be selected in order to get all the default configuration settings. Once you select the correct board, you will see that some configurations will be automatically selected, but be aware that some boards can have multiple versions (i.e different flash sizes).
+
+To select the board, go to ``Tools -> Board -> ESP32 Arduino`` and select the target board.
+
+If your board is not present on this list, you can select the generic ``ESP32-XX Dev Module``.
+
+Currently, we have one generic development module for each of the supported targets.
+
+If the board selected belongs to another SoC family, you will see the following information at the build output:
+
+    ``A fatal error occurred: This chip is ESP32 not ESP32-S2. Wrong --chip argument?``
+
+Upload Speed
+************
+
+To select the flashing speed, change the ``Tools -> Upload Speed``. This value will be used for flashing the code to the device.
+
+.. note:: If you have issues while flashing the device at high speed, try to decrease this value. This could be due to the external serial-to-USB chip limitations.
+
+CPU Frequency
+*************
+
+On this option, you can select the CPU clock frequency. This option is critical and must be selected according to the high-frequency crystal present on the board and the radio usage (Wi-Fi and Bluetooth).
+
+In some application, reducing the CPU clock frequency is recommended in order to reduce the power consumption.
+
+If you don't know why you shuld change this frequency, leave the default option.
+
+Flash Frequency
+***************
+
+Use this function to select the flash memory frequency. The frequency will be dependant of the memory model.
+
+* **40MHz**
+* **80MHz**
+
+If you don't know if your memory supports **80Mhz**, you can try to upload you scketch and watch the log output via the serial monitor.
+
+Flash Mode
+**********
+
+This option is used to select the SPI communication mode with the flash memory.
+
+Depending on the application, this mode can be changed in order to increase the flash communication speed.
+
+* **QIO** - Quad I/O Fast Read
+    * Four SPI pins are used to write to the flash and to read from flash.
+
+* **DIO** - Dual I/O Fast Read
+    * Two SPI pins are used to write to the flash and to read from flash.
+
+* **QOUT** - Quad Output Fast Read
+    * Four SPI pins are used to read the flash data.
+
+* **DOUT** - Dual Output Fast Read
+    * Two SPI pins are used to read flash data.
+
+If you don't know how the board flash is physically connected or the flash memory model, try the **QIO/QOUT** first and then **DIO/DOUT**.
```

Comment:
I think that testing DIO/DOUT first makes more sense as it uses a simpler pinout schema.
Some boards may be configured to DIO only and may reset with QIO. 
Testing QIO first may lead to issues in a first try, although I think it is very rare.

Reply:
I think the opposite. If you have QIO/QOUT support on your HW, you should try it first and take the advantage of it.
It's better to fail first sometimes :)

Reply:
Chances are that DIO/DOUT will work anyways. Let's ask them to try QIO 80MHz first, so that they can get the best performance

Reply:
Actually there are very few flash chips that might need DIO and I know of only one board that requires DOUT. On other hand we will not get issues caused by performance degradation

Reply:
I've tried some alternative flash memories on ESP32 and only one worked out in DOUT mode. I think it was some Cypress S25 something.